lib/Runtime/Base/WindowsGlobalizationAdapter.cpp

```diff
@@ -694,7 +699,153 @@ if (this->object) \
+        char icuLangTag[ULOC_FULLNAME_CAPACITY] = { 0 };
+        bool success = false;
+        UErrorCode error = U_ZERO_ERROR;
+        int32_t parsedLength = 0;
+        int32_t forLangTagResultLength = 0;
+        int32_t toLangTagResultLength = 0;
+
+        // Convert input language tag to a locale ID for use in uloc_toLanguageTag API.
+        forLangTagResultLength = uloc_forLanguageTag(reinterpret_cast<const char *>(inputLangTagUtf8), icuLocaleId, ULOC_FULLNAME_CAPACITY, &parsedLength, &error);
+        success = forLangTagResultLength && parsedLength && U_SUCCESS(error);
+        if (!success)
+        {
+            AssertMsg(false, "uloc_forLanguageTag failed");
+            goto cleanup;
+        }
+
+        // Try to convert icuLocaleId (locale ID version of input locale string) to BCP47 language tag, using strict checks
+        toLangTagResultLength = uloc_toLanguageTag(icuLocaleId, icuLangTag, ULOC_FULLNAME_CAPACITY, TRUE, &error);
+        success = toLangTagResultLength && U_SUCCESS(error);
+        if (!success)
+        {
+            if (error == U_ILLEGAL_ARGUMENT_ERROR)
+            {
+                AssertMsg(false, "uloc_toLanguageTag: error U_ILLEGAL_ARGUMENT_ERROR");
+            }
+            else
+            {
+                AssertMsg(false, "uloc_toLanguageTag: unexpected error (besides U_ILLEGAL_ARGUMENT_ERROR)");
+            }
+
+            goto cleanup;
+        }
+
+        *normalizedLength = utf8::DecodeUnitsIntoAndNullTerminateNoAdvance(normalized,
+            reinterpret_cast<const utf8char_t *>(icuLangTag), reinterpret_cast<utf8char_t *>(icuLangTag + toLangTagResultLength), utf8::doDefault);
+
+    cleanup:
+        HeapDeleteArray(inputLangTagUtf8SizeAllocated, inputLangTagUtf8);
+        inputLangTagUtf8 = nullptr;
+        return success ? S_OK : E_INVALIDARG;
+    }
+
+    //bool IcuIntlAdapter::ResolveLocaleLookup(_In_ ScriptContext *scriptContext, _In_ JavascriptString *locale, _Out_ char16 *resolved)
+    bool IcuIntlAdapter::ResolveLocaleLookup(_In_ ScriptContext *scriptContext, _In_z_ const char16 *locale, _Out_ char16 *resolved)
+    {
+        // FIXME (doilij): implement ResolveLocaleLookup
+        resolved[0] = '\0';
+        return false;
+    }
+
+    //bool IcuIntlAdapter::ResolveLocaleBestFit(_In_ ScriptContext *scriptContext, _In_ JavascriptString *locale, _Out_ char16 *resolved)
+    bool IcuIntlAdapter::ResolveLocaleBestFit(_In_ ScriptContext *scriptContext, _In_z_ const char16 *locale, _Out_ char16 *resolved)
+    {
+        // Note: the "best fit" matcher is implementation-defined, so it is okay to return the same result as ResolveLocaleLookup.
+        // TODO (doilij): implement a better "best fit" matcher
+        return ResolveLocaleLookup(scriptContext, locale, resolved);
+    }
+
+    int IcuIntlAdapter::GetUserDefaultLocaleName(_Out_ LPWSTR lpLocaleName, _In_ int cchLocaleName)
+    {
+        // FIXME (doilij): implement GetUserDefaultLocaleName
```

Comment:
can you please tag it as XPLAT-TODO ? or we already have the interface available for all the platforms?